Browser/Browser.aspx.cs

```diff
@@ -1474,10 +1475,22 @@ private void FillFolderTree(IFolderInfo currentFolderInfo)
 
             FoldersTree.Nodes.Add(folderNode);
 
-            var folders = FolderManager.Instance.GetFolders(currentFolderInfo);
+            // gets the list of folders the specified user has read permissions and transfrom into dictionary:
+            //      Key = parrent folder id 
+            //      Value = list of child folders
+            // this will let us possible to create folders tree much faster on a recursion below
+            var readOnlyFolders = FolderManager.Instance.GetFolders(UserController.Instance.GetCurrentUserInfo())
```

Comment:
For clarity, I think we should rename this to be "readableFolders" or otherwise.  From a long-term support perspective, the usage of "readOnlyFolders" implies that the user ONLY has read permissions, which isn't necessarily the case.  Another alternative would be "availableFolders"